osmdroid-android/src/main/java/org/osmdroid/util/SpeechBalloonHelper.java

```diff
@@ -0,0 +1,87 @@
+package org.osmdroid.util;
+
+import org.osmdroid.views.overlay.SpeechBalloonOverlay;
+
+/**
+ * Helper class for {@link SpeechBalloonOverlay}
+ * whose main interest is method {@link #compute(RectL, PointL, double, PointL, PointL)}
+ * @since 6.1.1
+ * @author Fabrice Fontaine
+ */
+public class SpeechBalloonHelper {
```

Comment:
should this class be all static functions and drop the private variables? it seems like it's more of a utility class than anything else

Reply:
It's definitely a utility class. I don't have strong views on Singleton vs. static classes.
In that case, the key is the optimization: not recreating the `PointL` for every `draw`.